### PR TITLE
Add org.kde.kdenlive

### PR DESCRIPTION
**Pull request by eszlari**

### CODE REVIEW

**Comment by nedrichards** [comment]
I'm not an expert, but is there a reason why this doesn't have some of the dependencies of the version in Flatpak KDE Apps? https://github.com/KDE/flatpak-kde-applications/blob/master/org.kde.kdenlive.json

@aleixpol @grulja as our local KDE community and Flatpak experts I'd appreciate your thoughts on this if you have time.

File: org.kde.kdenlive.json [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/gnome/sources/gtk+/2.24/gtk+-2.24.31.tar.xz",
+                    "sha256": "68c1922732c7efc08df4656a5366dcc3afdc8791513400dac276009b40954658"
+                }
+            ]
+        },
+        {
+            "name": "kdecoration",
```
Comment (aleixpol) [inline org.kde.kdenlive.json]: Who uses kdecoration? If it's generically required it should possibly go to the Sdk.

File: org.kde.kdenlive.json [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.kde.org/stable/plasma/5.11.4/kdecoration-5.11.4.tar.xz",
+                    "sha256": "35619d57ffe727eb624e67dc180279536e081955720dd67eea6751395a83706f"
+                }
+            ]
+        },
+        {
+            "name": "oxygen",
```
Comment (aleixpol) [inline org.kde.kdenlive.json]: kdenlive doesn't hardcode oxygen.

**Comment by aleixpol** [comment]
> I'm not an expert, but is there a reason why this doesn't have some of the dependencies of the version in Flatpak KDE Apps?

AFAIU, it does, it just has many others additionally.

**Comment by nedrichards** [comment]
@aleixpol you're right, I wasn't looking closely enough - however I'm glad we got you onto the review anyway!

**Comment by nedrichards** [comment]
bot, build org.kde.kdenlive

File: org.kde.kdenlive.json [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.videolan.org/x264/snapshots/x264-snapshot-20171223-2245-stable.tar.bz2",
+                    "sha256": "ee4ba2c3d1caf786ad66d3fa86bdc368a8f827318788a62a9c8444b2c35816bf"
+                }
+            ]
+        },
+        {
+            "name": "x265",
```
Comment (nedrichards) [inline org.kde.kdenlive.json]: In order to build on arm and aarch64 you're going to need to replicate the fixes here: https://github.com/flathub/org.pitivi.Pitivi/commit/548615c8a5d389197438997ce63d324d5c4b17b5

File: org.kde.kdenlive.json [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+                {
+                    "type": "shell",
+                    "commands": [
+                        "tar xf gtest.tar.gz --strip 1"
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "libsamplerate",
```
Comment (nedrichards) [inline org.kde.kdenlive.json]: In order to fix this on aarch64 you'll need to rm-configure: true and autoreconf e.g. https://github.com/flathub/shared-modules/blob/9cc549960171590d6a0736d762c083a51d1c5779/udev/udev-175.json

File: org.kde.kdenlive.json [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+        "/lib/cmake",
+        "/share/aclocal",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig"
+    ],
+    "modules": [
+        {
+            "name": "gtk2",
```
Comment (TingPing) [inline org.kde.kdenlive.json]: What is gtk2 for, theming?

Comment (eszlari) [reply org.kde.kdenlive.json]: It's needed by a MLT module:
https://github.com/mltframework/mlt/tree/master/src/modules/gtk2

File: org.kde.kdenlive.json (hunk 1 above)
Comment (eszlari) [reply org.kde.kdenlive.json]: It was needed for building oxygen.

**Comment by nedrichards** [comment]
bot, build org.kde.kdenlive

File: org.kde.kdenlive.json [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+                {
+                    "type": "shell",
+                    "commands": [
+                        "tar xf opencv_contrib.tar.gz --strip 1"
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "gavl",
```
Comment (nedrichards) [inline org.kde.kdenlive.json]: this needs the autoreconf dance as well. There also seems to be a slightly different compile error here for ARM

Comment (nedrichards) [reply org.kde.kdenlive.json]: Ah, I remember now - I saw the same issue in Pitivi which also uses gavl. It's something to do with libm. If you can work out how to fix it, all the better! If not, then it's OK to use a flathub.json file to ban the arm architecture for now.

**Comment by nedrichards** [comment]
bot, build org.kde.kdenlive

**Comment by nedrichards** [comment]
with the flathub.json arm won't be built, however I'm happy to work on that in a PR after we merge this.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.kde.kdenlive

**Pull request closed without merging**